bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -135,8 +135,11 @@ public InsteonDeviceHandler(Thing thing) {
     public void initialize() {
         config = getConfigAs(InsteonDeviceConfiguration.class);
 
+        updateStatus(ThingStatus.UNKNOWN);
+
         scheduler.execute(() -> {
-            if (getBridge() == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
```

Comment:
I see many first checks done in this thread that could have been done outside (before scheduler.execute) and so before setting the status to UNKNOWN. This would avoid status being set to UNKNOWN and then immediately after to OFFLINE/CONFIGURATION_ERROR.
Everything that is fast to execute should remain in initialize().
I have not studied in details the big content run in the thread but if a thread was created, I assume this is at least required for a part of the code (because it can take several seconds to run).

Reply:
@lolodomo, I was told by @kaikreuzer that this code needed to be inside of an thread when it was ported from the oh1 binding.  Why does it matter if it's first set to UNKNOWN then a thread is started up, it is in a UNKNOWN state.

Reply:
This is better if initialize() can return with the final status set.
If it can't because there is stuff to run that needs many seconds, it is ok to set it first to UNKNOW and then update it later in the created thread.
Having useless status changes (in your case UNKNOW and then immediately after OFFLINE/CONFIGURATION_ERROR) could be apparently avoided so why not avoid it.
I am just proposing a better implementation but the code as it is in your PR will also work, of course.

Reply:
Maybe I am not clear with what I am suggesting ?
Of course, the thread must be kept if it contains code that needs several seconds to run.
My suggestion is to move outside the thread all the checks that can be run quickly and lead to a status OFFLINE/CONFIGURATION_ERROR.
Let me know if this is not clear.

Reply:
Just to explain:
```
    public void initialize() {
        config = getConfigAs(InsteonDeviceConfiguration.class);

            Bridge bridge = getBridge();
            if (bridge == null) {
                String msg = "An Insteon network bridge has not been selected for this device.";
                logger.warn("{} {}", thing.getUID().getAsString(), msg);

                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
                return;
            }

            InsteonDeviceConfiguration config = this.config;
            if (config == null) {
                String msg = "Insteon device configuration is null.";
                logger.warn("{} {}", thing.getUID().getAsString(), msg);

                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
                return;
            }
            String address = config.getAddress();
            if (!InsteonAddress.isValid(address)) {
                String msg = "Unable to start Insteon device, the insteon or X10 address '" + address
                        + "' is invalid. It must be in the format 'AB.CD.EF' or 'H.U' (X10).";
                logger.warn("{} {}", thing.getUID().getAsString(), msg);

                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
                return;
            }

        updateStatus(ThingStatus.UNKNOWN);

        scheduler.execute(() -> {
```

Reply:
None of it takes a long time to run, there's just a bunch of initialization code. During the initial port from OH1, I was told to have all the initialization run inside of a thread. It's also cleaner this way to have everything encapsulated in the same place.

BTW, Here's the lifecycle for one of my 60+ devices:

```
2022-05-25 11:00:11.066 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from UNINITIALIZED to UNINITIALIZED (BRIDGE_UNINITIALIZED)
2022-05-25 11:00:11.574 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2022-05-25 11:00:11.649 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from INITIALIZING to UNKNOWN
2022-05-25 11:00:14.847 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from UNKNOWN to OFFLINE (BRIDGE_OFFLINE)
2022-05-25 11:00:24.879 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
```


Prior to this change it would have looked like:

```
2022-05-25 11:00:11.066 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from UNINITIALIZED to UNINITIALIZED (BRIDGE_UNINITIALIZED)
2022-05-25 11:00:11.574 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2022-05-25 11:00:14.847 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'insteon:device:home:22F688' changed from INITIALIZING to ONLINE
```

Reply:
Thread should be avoided if all the code can be run quickly. 

https://www.openhab.org/docs/developer/bindings/#the-thinghandler

Reply:
I'd rather leave as is, it keeps the code cleaner since its all together in the same code block.

Reply:
Also, according to the logs I provided, it takes about 3 seconds to initialize. So it makes sense to do in a separate thread, correct? This could also be considerably longer for others based on factors such as the number of Insteon devices to configure, hardware that OH is running on, etc.

Reply:
Thread will be used for example if you need to execute a HTTP request to authenticate or get data.

Reply:
After removing the thread, the UNKNOWN status is no more needed.

Reply:
What's here is because of review comments in the past from @kaikreuzer, now you want me to remove the thread? It takes at least 2-3 seconds to initialize as you can see from the log I provided.

Reply:
UNKNOWN state was added because of the thread as initialize must set a status before returning.

Reply:
> It takes at least 2-3 seconds to initialize

So some code is really consuming a lot of time. That is probably why Kai suggested the thread. 
So comeback to my initial suggestion and just extract what should run very fast like checking if configuration parameyers ate ok in particular.

Reply:
What is here works, and the configuration errors are very rare, but need to be covered. Since a thread is needed, I going to leave as is.